cli/remotebazel/remotebazel.go

```diff
@@ -303,45 +282,40 @@ func diffUntrackedFile(path string) (string, error) {
 	}
 	patch, err := runGit(args...)
 	if err != nil {
-		if _, ok := err.(*exec.ExitError); ok {
-			return patch, nil
+		// `git diff` returns exit code 1 if there is (valid) diff. Explicitly
+		// check for this case.
+		if !strings.Contains(patch, "diff --git") {
+			return "", err
 		}
-		return "", err
 	}
 
 	return patch, nil
 }
 
-func Config(path string) (*RepoConfig, error) {
-	repo, err := git.PlainOpenWithOptions(path, &git.PlainOpenOptions{DetectDotGit: true})
-	if err != nil {
-		return nil, status.WrapError(err, "open git repo")
-	}
-
+func Config() (*RepoConfig, error) {
 	remote, err := determineRemote()
 	if err != nil {
 		return nil, status.WrapError(err, "determine remote")
 	}
 	fetchURL := remote.url
 	log.Debugf("Using fetch URL: %s", fetchURL)
 
-	branch, commit, err := getBaseBranchAndCommit(repo)
+	remoteData, err := runGit("remote", "show", remote.name)
```

Comment:
Nit: This is not obvious, but it is making a network call to the remote server with `git remote show origin`. We need to add `-n` to avoid network calls. I suspect if we need the remote information, `git ls-remote origin` might give us richer information.

Reply:
https://github.com/git/git/blob/cc01bad4a9f566cf4453c7edd6b433851b0835e2/builtin/remote.c#L1322-L1329

Looking at this, it seems like both `git remote show` and `git ls-remote` would call `transport_get_remote_refs` to list all remote refs. So I think we might be able to refactor the code to only call `git ls-remote` once, then re-use the result.

Reply:
Refactored so the CLI makes a single `git ls-remote --symref origin` call